factorio_rcon/_factorio_rcon.py

```diff
@@ -630,16 +630,17 @@ async def send_commands(self, commands: Dict[T, str]) -> Dict[T, Optional[str]]:
             else:
                 results[id_map[response.id]] = response.body.rstrip()
         return results
-    
-    async def __aenter__(self) -> Self:
+
+    @handle_socket_errors(alive_socket_required=True)
```

Comment:
Since we can't have an async constructor, this breaks `async with AsyncRCONClient()` as it'll never be connected.
There are two options here: Either don't check it or call `connect()` inside aenter. I think calling connect here makes the most sense?

The downside is that the sync/async enters are then different

Reply:
I must admit that I don't get why someone might not want to connect it's client immediatly when using a context manager so why not just ignore the connect_on_init parameter and call connect() in  __enter__ too?

Reply:
I agree, that makes more sense. Both context handlers should probably check `self.rcon_socket` and self.rcon_failure`, if a socket exists and there's no failure we know it's already connected and can skip connecting